Homework 12

mrTwice opened the pull request.

Otshepenec committed:
copy classes from homework_10

Otshepenec committed:
Added the three classes: Saver, Loader, and ContactParser.

- The Saver class handles saving contacts to a file.
- The Loader class loads contacts from a file.
- The ContactParser class is used for parsing contacts from a string representation.

These classes represent the core functionality for working with contacts in the application.

sproshchaev commented on homework_12/src/main/java/ru/otus/basic/yampolskiy/utils/Loader.java:
```diff
@@ -0,0 +1,53 @@
+
+    private static final String DIRECTORY_PATH = "PhoneBook";
+
+    public static Map<String, Contact> loadFromFile() {
```
Метод желательно отрефакторить с целью упрощения и уменьшения числа строк. Часть логики можно вынести в приватные методы этого же класса

sproshchaev commented on homework_12/src/main/java/ru/otus/basic/yampolskiy/utils/ContactParser.java:
```diff
@@ -0,0 +1,34 @@
+import ru.otus.basic.yampolskiy.entities.PhoneNumber;
+
+public class ContactParser {
+    public static Contact parseContact(String fileContent) {
```
Метод рекомендуется сократить, вынеся часть логики в приватные методы класса

sproshchaev commented on homework_12/src/main/java/ru/otus/basic/yampolskiy/utils/ContactParser.java:
```diff
@@ -0,0 +1,34 @@
+            } else if (line.startsWith("lastName:")) {
+                lastName = line.substring("lastName:".length()).trim();
+            } else if (line.contains(":")) {
+                if (contact == null && firstName != null && lastName != null) {
```
У нас всегда будет `contact == null` (строка 11), поэтому эту проверку можно убрать из if:
```
if (firstName != null && lastName != null) {
```

Otshepenec committed:
Refactored classes to improve code readability and simplify file writing process.

Changes:
- Replaced manual byte buffer management with BufferedWriter and OutputStreamWriter for simpler and more efficient text file writing.
- Added error handling and logging for directory creation failures.
- Improved readability by using standard Java libraries and simplifying methods.
- Ensured proper handling of potential IOExceptions with informative error messages.
